### PR TITLE
MM-19790: Remove randomness in oauth test AuthData.

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
The ID field of Gitlab OAuth data was generated as a random number
between 1-1000. Unfortunately, this gave a 1 in 1 million chance of two
users generated this way colliding and meaning if you had two test users
in a test case they silently ended up actually being the exact same
user. We don't actually need to generate this randomly, so instead
change to having the test cases explicitly set the ID number so the
behaviour is as expected and completely deterministic.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-19790

### CODE REVIEW

File: app/user_test.go [hunk 1]
```diff
@@ -74,8 +73,7 @@ func TestCreateOAuthUser(t *testing.T) {
 	th := Setup(t).InitBasic()
 	defer th.TearDown()
 
-	r := rand.New(rand.NewSource(time.Now().UnixNano()))
-	glUser := oauthgitlab.GitLabUser{Id: int64(r.Intn(1000)) + 1, Username: "o" + model.NewId(), Email: model.NewId() + "@simulator.amazonses.com", Name: "Joram Wilander"}
+	glUser := oauthgitlab.GitLabUser{Id: 42, Username: "o" + model.NewId(), Email: model.NewId() + "@simulator.amazonses.com", Name: "Joram Wilander"}
```
Comment (cpoile) [inline app/user_test.go]: 42. Nice.

**Comment by grundleborg** [comment]
@cpoile good point - there isn't just 1 possible collision but 1,000 of them!

File: app/user_test.go [hunk 2]
```diff
@@ -253,8 +251,8 @@ func TestUpdateOAuthUserAttrs(t *testing.T) {
 
 	var user, user2 *model.User
 	var gitlabUserObj oauthgitlab.GitLabUser
-	user, gitlabUserObj = createGitlabUser(t, th.App, username, email)
-	user2, _ = createGitlabUser(t, th.App, username2, email2)
+	user, gitlabUserObj = createGitlabUser(t, th.App, 1, username, email)
```
Comment (marianunez) [inline app/user_test.go]: Given that the error in the ticket shows that it had issues updating the same email and username, should we also remove the randomness here of the email/username that relies on the `NewId` that still uses `uuid.NewRandom()`?

Comment (grundleborg) [reply app/user_test.go]: no, a UUID is random enough that we don't need to do that. If we can't rely on UUIDs not colliding, then Mattermost is completely broken anyway.

**Pull request merged**